src/OpenApi/src/OpenApiRouteHandlerBuilderExtensions.cs

```diff
@@ -46,37 +36,72 @@ public static RouteHandlerBuilder WithOpenApi(this RouteHandlerBuilder builder)
     /// <param name="builder">The <see cref="RouteHandlerBuilder"/>.</param>
     /// <param name="configureOperation">An <see cref="Func{T, TResult}"/> that returns a new OpenAPI annotation given a generated operation.</param>
     /// <returns>A <see cref="RouteHandlerBuilder"/> that can be used to further customize the endpoint.</returns>
-    public static RouteHandlerBuilder WithOpenApi(this RouteHandlerBuilder builder, Func<OpenApiOperation, OpenApiOperation> configureOperation)
+    public static TBuilder WithOpenApi<TBuilder>(this TBuilder builder, Func<OpenApiOperation, OpenApiOperation> configureOperation)
+        where TBuilder : IEndpointConventionBuilder
+    {
+        builder.Add(endpointBuilder => AddAndConfigureOperationForEndpoint(endpointBuilder, configureOperation));
+        return builder;
+    }
+
+    private static void AddAndConfigureOperationForEndpoint(EndpointBuilder endpointBuilder, Func<OpenApiOperation, OpenApiOperation>? configure = null)
     {
-        builder.Add(endpointBuilder =>
+        foreach (var item in endpointBuilder.Metadata)
         {
-            if (endpointBuilder is RouteEndpointBuilder routeEndpointBuilder)
+            if (item is OpenApiOperation existingOperation)
             {
-                var openApiOperation = GetOperationForEndpoint(routeEndpointBuilder);
-                if (openApiOperation != null)
+                if (configure is not null)
                 {
-                    routeEndpointBuilder.Metadata.Add(configureOperation(openApiOperation));
+                    var configuredOperation = configure(existingOperation);
```

Comment:
If I'm understanding this correctly, the callback provided to `WithOpenApi` will run on operations inserted by the parent group. So if I have something like:

```
var outer = app.MapGroup("/1").WithOpenApi();
var inner = outer.MapGroup("/2").WithOpenApi();
var nested = inner.MapGroup("/3").WithOpenApi(o => o.Summary = "foo");
var anotherOne = inner.MapGroup("/4").WithOpenApi();
```

Then the operation in the summary will be applied to the operations in `inner` and `outer`? What would `anotherOne` see in this case?

Reply:
Assuming that there was an endpoint added to `anotherOne` via `MapGet` or something, the path would be prefixed by "/1/2/4" (no 3 since `anotherOne` isn't under `nested`).

It would not see the "foo" summary since that was applied to the `nested`, but it would see anything done by callbacks to the `outer` and `inner` group. In this case, that's nothing.

Any endpoint defined under `nested` would have a "foo" summary unless overridden by a more nested group or on the endpoint specifically.